lxd/cluster/membership.go

```diff
@@ -468,24 +468,24 @@ func Join(state *state.State, gateway *Gateway, networkCert *shared.CertInfo, se
 				return fmt.Errorf("Failed to get storage pool driver: %w", err)
 			}
 
+			// For all pools we add the config provided by the joining node.
+			config, ok := pools[name]
+			if !ok {
+				return fmt.Errorf("Joining member has no config for pool %s", name)
+			}
+
+			err = tx.CreateStoragePoolConfig(id, node.ID, config)
+			if err != nil {
+				return fmt.Errorf("Failed to add joining node's pool config: %w", err)
+			}
+
 			if shared.StringInSlice(driver, []string{"ceph", "cephfs"}) {
 				// For ceph pools we have to create volume
 				// entries for the joining node.
 				err := tx.UpdateCephStoragePoolAfterNodeJoin(ctx, id, node.ID)
```

Comment:
@masnax on a separate topic, I don't like the look of this function call `UpdateCephStoragePoolAfterNodeJoin`.

Looking at its implementation it seems to be creating entries in storage_volumes for each member in the cluster.
But this isn't how its done for new entries created manually, my understanding is that storage_volumes on a remote (ceph) cluster have the node_id column value set to NULL (indicating it applies to all cluster members).

Please could you take a look if there is a difference between the storage_volumes table entries for a ceph pool when joining a cluster member vs creating the volume manually?